Python/import.c

```diff
@@ -387,11 +387,32 @@ import_get_module(PyThreadState *tstate, PyObject *name)
 }
 
 
-PyObject *
-PyImport_GetModule(PyObject *name)
+static int
+import_ensure_initialized(PyThreadState *tstate, PyObject *mod, PyObject *name)
 {
-    PyThreadState *tstate = _PyThreadState_GET();
-    return import_get_module(tstate, name);
+    PyInterpreterState *interp = tstate->interp;
+    PyObject *spec;
+
+    _Py_IDENTIFIER(__spec__);
+    _Py_IDENTIFIER(_lock_unlock_module);
+
+    /* Optimization: only call _bootstrap._lock_unlock_module() if
+       __spec__._initializing is true.
+       NOTE: because of this, initializing must be set *before*
+       stuffing the new module in sys.modules.
+    */
+    spec = _PyObject_GetAttrId(mod, &PyId___spec__);
```

Comment:
`_PyModuleSpec_IsInitializing()` is a little too clever IMHO, so lets make things a little more obvious here:

```suggestion
    spec = _PyObject_GetAttrId(mod, &PyId___spec__);
    if (spec == NULL) {
        PyErr_Clear();
        return 0;
    }
```

Reply:
`_PyModuleSpec_IsInitializing()` does this. I do not see reason to duplicate code.